qcore/constants.py

```diff
@@ -151,38 +232,86 @@ class ProcessType(ExtendedStrEnum):
         True,
         "{run_command} python $gmsim/workflow/workflow/calculation/bb_sim.py {outbin_dir} {vel_mod_dir} {hf_bin_path} {stat_vs_est} "
         "{bb_bin_path} --flo {flo}",
-        (1, 4),
+        (
+            (
+                Dependency(EMOD3D, DependencyTarget.REL),
+                Dependency(HF, DependencyTarget.REL),
+            ),
+        ),
     )
+    LF2BB = (
+        12,
+        "LF2BB",
+        None,
+        None,
+        None,
+        ((Dependency(EMOD3D, DependencyTarget.REL),),),
+    )
+    HF2BB = 13, "HF2BB", None, None, None, ((Dependency(HF, DependencyTarget.REL),),)
     IM_calculation = (
         6,
         "IM_calc",
         False,
         False,
         "time python $IMPATH/calculate_ims.py {sim_dir}/BB/Acc/BB.bin b -o {sim_dir}/IM_calc/ -np {np} -i "
         "{sim_name} -r {fault_name} -t s {component} {extended} {simple} {advanced_IM} {pSA_periods}",
-        ((5,), (12,), (13,)),
+        (
+            (Dependency(BB, DependencyTarget.REL),),
+            (Dependency(LF2BB, DependencyTarget.REL),),
+            (Dependency(HF2BB, DependencyTarget.REL),),
+        ),
     )
```

Comment:
I have a bit of problem with the readability.
```
    BB = (
        5,
...
        (
            (
                Dependency(EMOD3D, DependencyTarget.REL),
                Dependency(HF, DependencyTarget.REL),
            ),
        ),
    )
```
and
```
    IM_calculation = (
        6,
        "IM_calc",
...
        (
            (Dependency(BB, DependencyTarget.REL),),
            (Dependency(LF2BB, DependencyTarget.REL),),
            (Dependency(HF2BB, DependencyTarget.REL),),
        ),
    )
```
For BB, both dependencies should be met, and IM_calculation, either of 3 can be met. 
Here, the multiple layers of tuples makes things quite confusing and less intuitive. Also, I find it a bit too verbose.

Can we make it something like

```
    BB = (
        5,
...
        (
            [EMOD3D.REL, HF.REL]
        ),
    )
```
and

```
    IM_calculation = (
        6,
     ...
        (
            [BB.REL],
            [LF2BB.REL],
            [HF2BB.REL],
        ),
    )
```
where [] used for all and () used for any. (Either way is fine) 
Regarding the dot in LF2BB.REL, it's just an example of possible simplification.  I am sure there is a better alternative notation.